### ISSUE TITLE
zero-shot inference and inverse folding with fixed residues

### ISSUE THREAD

**Issue by adrienchaton** [microsoft/protein-sequence-models#13]
Hello,

Thank you for sharing your model.
I have a few questions on how to best use MIF-ST please.

In your example script, the model is provided with features derived from the PDB coordinates and sequence ("wt").

For zero-shot inference of e.g. single point mutants, should we replace the input sequence "wt" with a mask token "#" at the mutated residue for computing the MMP score as in ESM (only taking the logits at that residue and computing the difference of log probabilities) ? Or should we forward through the model coords+wt and coords+mutant to score them independently with e.g. average log-likelihood over all residue logits ?

For inverse folding, e.g. redesigning certain residues in a sequence, can I use MIF-ST ? And would it be reasonable to mask all the residues to mutate at once, forward through the model coords+masked_seq and then sample from the logits at the masked residues ?

Thanks !

**Comment by yangkky** [comment]
> For zero-shot inference of e.g. single point mutants, should we replace the input sequence "wt" with a mask token "#" at the mutated residue for computing the MMP score as in ESM (only taking the logits at that residue and computing the difference of log probabilities) ? Or should we forward through the model coords+wt and coords+mutant to score them independently with e.g. average log-likelihood over all residue logits ?

Either of those methods should work. For the paper, we use the former method. 

> For inverse folding, e.g. redesigning certain residues in a sequence, can I use MIF-ST ? And would it be reasonable to mask all the residues to mutate at once, forward through the model coords+masked_seq and then sample from the logits at the masked residues ?

I haven't tried this, but the best method would probably be some sort of Gibbs sampling, as in this [paper](https://www.biorxiv.org/content/10.1101/2021.01.26.428322v1)

**Comment by adrienchaton** [comment]
Thanks for your reply !
